contracts/src/Token.ts

```diff
@@ -75,4 +69,26 @@ export class ExampleToken extends SmartContract {
       amount,
     });
   }
+
+  @method deployZkapp(address: PublicKey, verificationKey: VerificationKey) {
+    let tokenId = this.token.id;
+    let zkapp = AccountUpdate.create(address, tokenId);
+    zkapp.account.permissions.set(Permissions.default());
+    zkapp.account.verificationKey.set(verificationKey);
+    zkapp.requireSignature();
+  }
+
+  @method approveUpdateAndSend(
```

Comment:
Not sure about this function. It is used to send tokens from user1 to user2, but instead just mints new tokens and sends them to user2. Doesn't seem right

Reply:
See https://github.com/ycryptx/recursive-zk-lightning-network/pull/4#discussion_r1176056729, it is to send token from zkApp. It is a quite standardized way to do it from zkApp. For relevant information, see https://github.com/o1-labs/snarkyjs/blob/main/src/examples/zkapps/dex/dex.ts#L429